apps/ledger-live-mobile/src/newArch/features/FirmwareUpdate/components/UpdateBanner/UpdateBanner.test.tsx

```diff
@@ -201,11 +207,98 @@ describe("<UpdateBanner />", () => {
     expect(navigateToNewUpdateFlow).not.toHaveBeenCalled();
   });
 
+  it("should open the unsupported drawer if there is a bluetooth update on Android on Nano X with version < 2.4.0", async () => {
+    PlatformSpy.mockReturnValue({ OS: "android" } as typeof ReactNative.Platform);
+    useLatestFirmware.mockReturnValue({
+      final: {
+        name: "mockVersion",
+        version: "2.4.0",
+      },
+    });
+
+    const mockDeviceModelId = DeviceModelId.nanoX;
+    const mockDeviceVersion = "2.3.0";
+    const { user } = render(<UpdateBanner onBackFromUpdate={() => {}} />, {
+      overrideInitialState: makeOverrideInitialState({
+        deviceModelId: mockDeviceModelId,
+        version: mockDeviceVersion,
+        hasCompletedOnboarding: true,
+        wired: false,
+        hasConnectedDevice: true,
+      }),
+    });
+
+    // Check that the banner is displayed with the correct wording
+    expect(await screen.findByText("OS update available")).toBeOnTheScreen();
+    expect(
+      await screen.findByText("Tap to update your Ledger Nano X to OS version mockVersion."),
+    ).toBeOnTheScreen();
+
+    // Press the banner
+    await user.press(screen.getByTestId("fw-update-banner"));
+
+    // Check that the unsupported drawer is displayed
+    expect(await screen.findByText("USB cable needed")).toBeOnTheScreen();
+    expect(
+      await screen.findByText(
+        "To start the firmware update, plug your Ledger Nano X to your mobile phone using a USB cable.",
+      ),
+    ).toBeOnTheScreen();
+
+    // Check that the entrypoints to the update flows are not called
+    expect(navigateToOldUpdateFlow).not.toHaveBeenCalled();
+    expect(navigateToNewUpdateFlow).not.toHaveBeenCalled();
+  });
+
+  it("should open the unsupported drawer if there is a bluetooth update on iOS on Nano X with version < 2.4.0", async () => {
+    PlatformSpy.mockReturnValue({ OS: "ios" } as typeof ReactNative.Platform);
+    useLatestFirmware.mockReturnValue({
+      final: {
+        name: "mockVersion",
+        version: "2.4.0",
+      },
+    });
+
+    const mockDeviceModelId = DeviceModelId.nanoX;
+    const mockDeviceVersion = "2.3.0";
+    const { user } = render(<UpdateBanner onBackFromUpdate={() => {}} />, {
+      overrideInitialState: makeOverrideInitialState({
+        deviceModelId: mockDeviceModelId,
+        version: mockDeviceVersion,
+        hasCompletedOnboarding: true,
+        wired: false,
+        hasConnectedDevice: true,
+      }),
+    });
```

Comment:
As this is shared between both we could create a const above it that we can reuse for the test. And if only one or two values change we can override it depending on the need. It improves the readability of the test by extracting everything that is needed and you quickly see what's specific to the use case you're covering by the test. wdyt?